whois-nrtm4/src/main/java/net/ripe/db/nrtm4/NrtmFileProcessor.java

```diff
@@ -21,38 +21,38 @@ public class NrtmFileProcessor {
 
     private final NrtmFileService nrtmFileService;
     private final NrtmProcessControl nrtmProcessControl;
-    private final NrtmSourceHolder nrtmSourceHolder;
     private final SnapshotFileGenerator snapshotFileGenerator;
+    private final SourceRepository sourceRepository;
 
     public NrtmFileProcessor(
         final NrtmFileService nrtmFileService,
         final NrtmProcessControl nrtmProcessControl,
-        final NrtmSourceHolder nrtmSourceHolder,
-        final SnapshotFileGenerator snapshotFileGenerator
+        final SnapshotFileGenerator snapshotFileGenerator,
+        final SourceRepository sourceRepository
     ) {
         this.nrtmFileService = nrtmFileService;
         this.nrtmProcessControl = nrtmProcessControl;
-        this.nrtmSourceHolder = nrtmSourceHolder;
         this.snapshotFileGenerator = snapshotFileGenerator;
+        this.sourceRepository = sourceRepository;
     }
 
     public void updateNrtmFilesAndPublishNotification() throws IOException {
-        LOGGER.info("runWrite() called");
-        final NrtmSource source = nrtmSourceHolder.getSource();
-        final Optional<SnapshotFile> lastSnapshot = snapshotFileGenerator.getLastSnapshot(source);
-        Optional<PublishableSnapshotFile> publishableSnapshotFile = Optional.empty();
+        LOGGER.info("updateNrtmFilesAndPublishNotification() called");
+        final List<NrtmSourceModel> sourceList = sourceRepository.getAllSources();
+        if (sourceList.isEmpty()) {
+            sourceRepository.createSources();
+        }
+        final Optional<SnapshotFile> lastSnapshot = snapshotFileGenerator.getLastSnapshot(sourceRepository.getSource().orElseThrow());
         if (lastSnapshot.isEmpty()) {
```

Comment:
Is this checking for whois.source only? What will happen if non auth source does not exists?  So either you check both sources to decide if initialisation is needed or not or you check if just the tables is empty or not? But I would prefer you just check for empty table as your initialisation logic expects an empty database . correct?

Reply:
as far as I understand we create everything for all the sources at same time. So if we check one of the source and that source contains the snapshot it means that the rest of them contains the snapshot as well. This check should work correctly and if it doesn't is because be have a bug. Therefore I really don't mind to check if the source has the snapshot or if this table is empty or not (regardless the source)